src/services/transaction_watcher_signer_service.ts

```diff
@@ -0,0 +1,364 @@
+            throw new Error(`no signer available with this publicAddress: ${publicAddress}`);
+        }
+        return signer;
+    }
+    private async _getNextSignerAsync(): Promise<Signer> {
+        const sortedSigners = await this._getSortedSignerPublicAddressesByAvailabilityAsync();
+        // TODO(oskar) - add random choice for top signers to better distribute
+        // the fees.
+        const signer = this._signers.get(sortedSigners[0]);
+        if (signer === undefined) {
+            throw new Error(`signer with public address: ${sortedSigners[0]} is not available`);
+        }
+
+        return signer;
+    }
+    private async _getSortedSignerPublicAddressesByAvailabilityAsync(): Promise<string[]> {
+        const map = new Map<string, number>();
+        this._availableSignerPublicAddresses.forEach(signerAddress => {
+            map.set(signerAddress, 0);
+        });
+        // TODO(oskar) - move to query builder?
+        const res: Array<{ from: string; count: number }> = await this._transactionRepository.query(
+            `SELECT transactions.from, COUNT(*) FROM transactions WHERE status in ('submitted','mempool','stuck') GROUP BY transactions.from`,
+        );
+        res.filter(result => {
+            // we exclude from addresses that are not part of the available
+            // signer pool
+            return this._availableSignerPublicAddresses.includes(result.from);
+        }).forEach(result => {
+            map.set(result.from, result.count);
+        });
+        return [...map.entries()]
+            .sort((a, b) => {
+                return a[1] - b[1];
+            })
+            .map(entry => entry[0]);
+    }
+    private async _unstickTransactionAsync(
+        tx: TransactionEntity,
+        gasPrice: BigNumber,
+        signer: Signer,
+    ): Promise<string> {
+        if (tx.nonce === undefined) {
+            throw new Error(`failed to unstick transaction ${tx.txHash} nonce is undefined`);
+        }
+        const txHash = await signer.sendTransactionToItselfWithNonceAsync(tx.nonce, gasPrice);
+        const transactionEntity = TransactionEntity.make({
+            refHash: txHash,
+            txHash,
+            status: TransactionStates.Submitted,
+            nonce: tx.nonce,
+            gasPrice,
+            from: tx.from,
+            expectedMinedInSec: EXPECTED_MINED_SEC,
+        });
+        await this._transactionRepository.save(transactionEntity);
+        return txHash;
+    }
+    private async _checkForStuckTransactionsAsync(): Promise<void> {
+        const stuckTransactions = await this._transactionRepository.find({
+            where: { status: TransactionStates.Stuck },
+        });
+        if (stuckTransactions.length === 0) {
+            return;
+        }
+        const gasStationPrice = await this._getGasPriceFromGasStationOrThrowAsync();
+        const targetGasPrice = gasStationPrice.multipliedBy(UNSTICKING_TRANSACTION_GAS_MULTIPLIER);
```

Comment:
suggestion: lower `UNSTICKING_TRANSACTION_GAS_MULTIPLIER` to like 1.1, fast should already be fast enough, no point in  paging `20 * 1.5` to unstick it if 20 should do the job (eventually)